stages/cloud-init: add new stage for configuring cloud-init

thozza opened the pull request:
Add new `org.osbuild.cloud-init` stage, which currently allows to create configuration files for cloud-init under `/etc/cloud/cloud.cfg.d`. The stage` supports only a very limited subset of cloud-init configuration options, which is covering needs of RHEL AMI images.

The schema mandates that if the 'configuration_files' option is specified, then at least one configuration file must be defined. In addition each section of the configuration must contain at least one property (section or configuration option).

Add `python3-pyyaml` package to the `F34-build` testing manifest, because it is required for running and testing the new stage.

Regenerate all affected manifests.

Add test for the new stage.

gicmo reviewed (approved):
Cool, thanks! I had to look twice to understand what is going on because it is so simple and just writes the options!  Maybe comment, but I think it is fine! 

NB: The more I look at that `"configuration_files"` the weirder I find it, but we have it now in other stages, so let's keep it here as well.

thozza commented:
> Cool, thanks! I had to look twice to understand what is going on because it is so simple and just writes the options! Maybe comment, but I think it is fine!

Oh yeah, I was surprised myself how short and simple it was 😄 

> NB: The more I look at that `"configuration_files"` the weirder I find it, but we have it now in other stages, so let's keep it here as well.

As we discussed over Slack, I'll keep it as is for now. I'll be happy to help make it consistent in the future if needed!

thozza commented:
I added a short comment for the `create_configuration_file()`, explaining why it is so simple and rebased the PR on top of the latest `main`. There were no additional changes.

gicmo reviewed (approved):
👍

thozza commented:
Ehm... I had to regenerate test case data for the previously merged `dracut.conf` stage due to added `python3-pyyaml` into the `f34-build.json` manifest in this PR...

gicmo reviewed (approved):
Still good!
